packages/medusa/src/types/sales-channels.ts

```diff
@@ -1,7 +1,12 @@
+import { Request } from "express"
+import { SalesChannel } from "../models/sales-channel"
+
 export type CreateSalesChannelInput = {
   name: string
   description?: string
   is_disabled?: boolean
 }
 
 export type UpdateSalesChannelInput = Partial<CreateSalesChannelInput>
+
+export type SalesChannelRequest = Request & { sales_channel: SalesChannel }
```

Comment:
thought: Would it make sense to make a more generic type to use for all request types?

Something along the lines of:
```javascript
export type ExtendedRequest<TEntity> = Request & TEntity 
```

Reply:
it would be very nice, but we don't want to allow the entity properties in the request, I would suggest maybe something like that
```ts
export type ExtendedRequest<TEntity> = Request & { resource: TEntity }
```

Which is a minor change to your proposal :)

Reply:
Makes sense! I like this typing, but this decision can be considered a code convention, so would like to have everything aligned on it

Reply:
Yeah sure, I added the type to be used here, I can write a convention on that tomorrow morning if we go with that. and in a different pr I can update the batch job to use that new typings/assignment